### PR TITLE
zoom: Add registry entries

### PR DESCRIPTION
**Pull request by klaidliadon**
It adds a registry file which register the application to open zoom links.

The URL of the reg files should be in the main repo instead of the fork.

### CODE REVIEW

File: scripts/zoom-uninstall-context.reg [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Windows Registry Editor Version 5.00
+
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher\shell]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher\shell\open]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher\shell\open\command]
```
Comment (Ash258) [inline scripts/zoom-uninstall-context.reg:6]:
```suggestion
```

File: scripts/zoom-uninstall-context.reg [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\DefaultIcon]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\shell]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\shell\open]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\shell\open\command]
```
Comment (Ash258) [inline scripts/zoom-uninstall-context.reg:11]:
```suggestion
```

File: scripts/zoom-uninstall-context.reg [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\DefaultIcon]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\shell]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\shell\open]
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\shell\open\command]
```
Comment (Ash258) [inline scripts/zoom-uninstall-context.reg:16]:
```suggestion
```

File: scripts/zoom-uninstall-context.reg [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+[-HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPhoneCall]
+[-HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\Shell\Associations\UrlAssociations\IM]
+[-HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\Shell\Associations\UrlAssociations\IM\UserChoice]
+[-HKEY_CURRENT_USER\SOFTWARE\RegisteredApplications]
```
Comment (Ash258) [inline scripts/zoom-uninstall-context.reg:20]: Why would you remove full key??

Delete just zoom

File: scripts/zoom-install-context.reg [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\DefaultIcon]
+@="\"$code\",1"
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\shell]
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\shell\open]
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\shell\open\command]
+@="\"$code\" --url=\"%l\""
```
Comment (Ash258) [inline scripts/zoom-install-context.reg:41]: Code??

File: scripts/zoom-install-context.reg [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+[HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\shell\open\command]
+@="\"$code\" \"--url=%1\""
+
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im]
+@="URL:IM Protocol"
+"URL Protocol"=""
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomPbx.im\DefaultIcon]
+@="\"$code\",1"
```
Comment (Ash258) [inline scripts/zoom-install-context.reg:34]: Code??

File: scripts/zoom-install-context.reg [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\DefaultIcon]
+@="\"$code\",1"
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\shell]
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\shell\open]
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\shell\open\command]
+@="\"$code\" \"--url=%1\""
```
Comment (Ash258) [inline scripts/zoom-install-context.reg:26]: Code??

File: scripts/zoom-install-context.reg [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher\shell\open\command]
+@="\"$code\" \"--url=%1\""
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg]
+"URL Protocol"=""
+@="URL:Zoom Launcher"
+"UseOriginalUrlEncoding"=dword:00000001
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\zoommtg\DefaultIcon]
+@="\"$code\",1"
```
Comment (Ash258) [inline scripts/zoom-install-context.reg:19]: Code??

File: scripts/zoom-install-context.reg [hunk 9]
```diff
@@ -0,0 +1,52 @@
+Windows Registry Editor Version 5.00
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher]
+@="Zoom Launcher - 3.0.1"
```
Comment (Ash258) [inline scripts/zoom-install-context.reg:4]: Hardcoded version??

File: scripts/zoom-install-context.reg [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher\shell]
+
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher\shell\open]
+
```
Comment (Ash258) [inline scripts/zoom-install-context.reg:9]:
```suggestion
```

File: scripts/zoom-install-context.reg [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+[HKEY_CURRENT_USER\SOFTWARE\Classes\ZoomLauncher\shell\open\command]
+@="\"$code\" \"--url=%1\""
```
Comment (Ash258) [inline scripts/zoom-install-context.reg:11]: Code??

File: bucket/zoom.json [hunk 12]
```diff
@@ -6,8 +6,17 @@
         "identifier": "Proprietary",
         "url": "https://zoom.us/terms"
     },
-    "url": "https://zoom.us/client/latest/ZoomInstaller.exe#/dl.7z",
-    "hash": "e3b464d37e4b8d00517df822474b473bf8eccb308e4d35a0601d0be4646ff559",
+    "notes": "Add Visual Studio Code as a context menu option by running: \"$dir\\zoom-install-context.reg\"",
+    "url": [
+        "https://zoom.us/client/latest/ZoomInstaller.exe#/dl.7z",
+        "https://raw.githubusercontent.com/klaidliadon/scoop-extras/master/scripts/zoom-install-context.reg",
+        "https://raw.githubusercontent.com/klaidliadon/scoop-extras/master/scripts/zoom-uninstall-context.reg"
```
Comment (Ash258) [inline bucket/zoom.json:13]: Create nested folder zoom in scripst

File: bucket/zoom.json [hunk 13]
```diff
@@ -19,6 +28,24 @@
             "Zoom"
         ]
     ],
+    "post_install": [
+        "if (Test-Path \"$dir\\zoom-install-context.reg\") {",
+        "  $codepath = \"$dir\\Zoom.exe\".Replace('\\', '\\\\')",
```
Comment (Ash258) [inline bucket/zoom.json:33]: Rename variable

File: bucket/zoom.json [hunk 14]
```diff
@@ -6,8 +6,17 @@
         "identifier": "Proprietary",
         "url": "https://zoom.us/terms"
     },
-    "url": "https://zoom.us/client/latest/ZoomInstaller.exe#/dl.7z",
-    "hash": "e3b464d37e4b8d00517df822474b473bf8eccb308e4d35a0601d0be4646ff559",
+    "notes": "Add Visual Studio Code as a context menu option by running: \"$dir\\zoom-install-context.reg\"",
```
Comment (Ash258) [inline bucket/zoom.json:9]: Code??

**Comment by Ash258** [comment]
Why do you request review whey you did not finish the previous one and when you cannot answer the question....

**Comment by klaidliadon** [comment]
Every comment should be addressed now

**Comment by Ash258** [comment]
No it is not... At minimum 4 first requested changes are still not in place

https://github.com/lukesampson/scoop-extras/pull/4055#discussion_r426133627

**Pull request closed without merging**